enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,1319 @@
+
+In this enhancement we would be adding API fields in the installer and the Ingress Controller CR
+to set the AWS EIP for the NLB Ingress Controller. The cluster ingress operator will then create 
+the ingress controller and get the EIP assigned to the service LoadBalancer with the annotation 
+`service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+Note: `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` is a [kubernetes service annotation](https://kubernetes.io/docs/reference/labels-annotations-taints/#service-beta-kubernetes-io-aws-load-balancer-eip-allocations).
+
+
+### Workflow Description
+
+Pre-requisites: One must [allocate Elastic IP address](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html#using-instance-addressing-eips-allocating)
+in their AWS environment which could be then used in the `eipAllocations` field of the IngressController CR.
+Note: The AWS account has some cap on the number of EIPs per region. By default, all AWS accounts have a quota of five (5) Elastic IP addresses per Region, because public (IPv4) 
+internet addresses are a scarce public resource. For more details please check [Elastic IP address quota](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html#using-instance-addressing-limit)
+When this cap is hit you will get the following error: `Elastic IP address could not be allocated.
+The maximum number of addresses has been reached.`
+
+**cluster administrator** is a human user responsible for deploying a
+cluster. A cluster administrator also has the rights to modify the cluster level components.
+
+#### Set EIP using an installer:
+1. Cluster administrator creates `install-config.yaml` and adds the following in it. 
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  aws:
+    region: us-west-2
+    lbType: NLB
+      networkLoadBalancer:
+        eipAllocations:
+        - eipalloc-1111
+        - eipalloc-2222
+        - eipalloc-3333
+        - eipalloc-4444
+        - eipalloc-5555 
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```  
+Here, the administrator needs to know how many public subnets the VPC ID of their cluster has.
+They may check it by clicking the `Load Balancers` tab then under the details click `VPC ID` under VPC, then click the link under `Main route table` and then check the column `Explicit subnet associations` which
+will show you the number of public subnets available for that VPC ID.
+
+Other way to check VPC ID and subnet is go `Your VPCs` under `Virtual Public CLoud` section, then search for the cluster name using the
+command `oc get infrastructure cluster -o jsonpath='{.status.infrastructureName}'`
+
+One can try these scripts present [here](https://github.com/frobware/haproxy-hacks/tree/master/NE-1398) as well.
+
+2. The installer will create the cluster ingress config object as follows:
+```yaml
+ oc edit ingress.config/cluster -o yaml
```

Comment:
Can you clean up the command (use `oc get`) and yaml so that only necessary items are shown?  e.g. remove creationTimestamp, generation, resourceVersion, uid.  Fix domain if it should match the previous example yaml.

Reply:
yes